core/resources.rs

```diff
@@ -23,9 +25,50 @@ use std::rc::Rc;
 /// Returned by resource read/write/shutdown methods
 pub type AsyncResult<T> = Pin<Box<dyn Future<Output = Result<T, Error>>>>;
 
-/// All objects that can be store in the resource table should implement the
-/// `Resource` trait.
-/// TODO(@AaronO): investigate avoiding alloc on read/write/shutdown
+/// Resources are Rust objects that are attached to a [deno_core::JsRuntime].
+/// They live at most as long as the runtime lives. They are identified in JS
```

Comment:
I'm not sure if they live at most as long as the runtime - since most resources are `Rc<>` it's possible that caller might be holding to one of these even when runtime is dropped?